src/main/java/com/puppycrawl/tools/checkstyle/utils/AnnotationUtil.java

```diff
@@ -123,20 +123,88 @@ public static boolean containsAnnotation(DetailAST ast, Set<String> annotations)
 
         if (!annotations.isEmpty()) {
             final DetailAST firstMatchingAnnotation = findFirstAnnotation(ast, annotationNode -> {
-                DetailAST identNode = annotationNode.findFirstToken(TokenTypes.IDENT);
-                if (identNode == null) {
-                    identNode = annotationNode.findFirstToken(TokenTypes.DOT)
-                            .findFirstToken(TokenTypes.IDENT);
-                }
-
-                return annotations.contains(identNode.getText());
+                return doesAnnotationMatch(annotationNode, annotations);
```

Comment:
Can we create a FullIdent here, then simply check if `annotations` contains this string?

Reply:
> Can we create a FullIdent here

I think no. The null check in `doesAnnotationMatch()` is needed. Straight up passing `annotationNode` to `FullIdent#createFullIdent()` does not work as expected in all cases.

> then simply check if `annotations` contains this string?

The purpose of the `doesAnnotationMatch()` is for readability purposes. Without it `containsAnnotation()` becomes a bit hard to look at in my opinion.

Reply:
Ok, please push your changes and we will take a look. Note that `Fullident#createFullIdent` is essentially null-safe.